src/gluonts/model/estimator.py

```diff
@@ -83,7 +87,9 @@ class DummyEstimator(Estimator):
     def __init__(self, predictor_cls: type, **kwargs) -> None:
         self.predictor = predictor_cls(**kwargs)
 
-    def train(self, training_data: Dataset) -> Predictor:
+    def train(
+        self, training_data: Dataset, valid_dataset: Optional[Dataset] = None
```

Comment:
This is a breaking change, and the more that I think about it, the more I dislike our overall design of estimators and predictors. Essentially, such a change shouldn't be breaking.

We have models that don't require training data, i.e. local models which "train" online. Still, if we want an estimator, we would have to implement a train method for them which takes a training dataset as input.

I think this approach is another step in this direction. Now we add another dataset to `train`, and some models use it and others don't. 

Our design also inhabits the "class with only one method" code-smell, albeit there might be some good reasons to do it this way.

I'm also wondering if our current approach limits our future design space. E.g., if we want to develop models for anomaly detection, we might have different inputs than for normal forecasting models.

This is not to take anything away from your great work @strawberrypie, but I think I would like to have this discussion rather now then later.

/cc @vafl @lostella @jgasthaus

Reply:
@jaheba I feel your pain and agree that the interaction between components here is a bit weird

However, before we dive into the details of this larger issue:
- How is this change breaking? Note that the validation dataset here is optional
- It seems to me that with this change all estimators would support validation data during training

Reply:
There are estimators outside of GluonTS ;)

These would also need to be adapted, since they now need to take an additional argument.

Reply:
How about we merge this PR and we discuss that topic at a later point?